pkg/operator/staticpod/controller/revision/revision_controller.go

```diff
@@ -129,29 +134,47 @@ func nameFor(name string, revision int32) string {
 
 // isLatestRevisionCurrent returns whether the latest revision is up to date and an optional reason
 func (c RevisionController) isLatestRevisionCurrent(revision int32) (bool, string) {
-	for _, name := range c.configMaps {
-		required, err := c.configMapGetter.ConfigMaps(c.targetNamespace).Get(name, metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
+	for _, cm := range c.configMaps {
+		requiredData := map[string]string{}
+		existingData := map[string]string{}
+
+		required, err := c.configMapGetter.ConfigMaps(c.targetNamespace).Get(cm.Name, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) && !cm.Optional {
 			return false, err.Error()
 		}
-		existing, err := c.configMapGetter.ConfigMaps(c.targetNamespace).Get(nameFor(name, revision), metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
+		existing, err := c.configMapGetter.ConfigMaps(c.targetNamespace).Get(nameFor(cm.Name, revision), metav1.GetOptions{})
+		if apierrors.IsNotFound(err) && !cm.Optional {
 			return false, err.Error()
```

Comment:
not new, but this should probably be `return false, nil`, right?  otherwise I can't easily add anything.

Reply:
> t new, but this should probably be `return false, nil`, right? otherwise I can't easily add anything.

never mind.  reason, not error. ignore me